pkg/backend/aws/k8s_compute_backend.go

```diff
@@ -0,0 +1,121 @@
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+	"github.com/chanzuckerberg/happy/pkg/backend/aws/interfaces"
+	"github.com/chanzuckerberg/happy/pkg/config"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/homedir"
+)
+
+type K8SComputeBackend struct {
+	Backend     *Backend
+	ClientSet   *kubernetes.Clientset
+	HappyConfig *config.HappyConfig
+}
+
+func NewK8SComputeBackend(ctx context.Context, happyConfig *config.HappyConfig, b *Backend) (interfaces.ComputeBackend, error) {
+	var rawConfig *rest.Config
+	if happyConfig.K8SConfig().AuthMethod == "eks" {
+		// Constructs client configuration dynamically
+		clusterId := happyConfig.K8SConfig().ClusterID
+
+		clusterInfo, err := b.eksclient.DescribeCluster(context.TODO(), &eks.DescribeClusterInput{
+			Name: &clusterId,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to get k8s cluster configuration")
+		}
+		logrus.Infof("EKS Authenticated K8S Cluster: %s (%s)\n", *(clusterInfo.Cluster).Name, *(clusterInfo.Cluster).Version)
```

Comment:
I think these types of logs should be debug so they don't show up in the normal flow of the CLI. And if you want to see the more verbose logs, use `-v`. This log is more for us as developers of happy and not our customers.